src/flynt/transform/percent_transformer.py

```diff
@@ -53,6 +53,32 @@ def formatted_value(
         fmt_prefix = fmt_prefix.replace(".", "0")
 
     if fmt_spec in conversion_methods:
+        if fmt_spec == "s" and fmt_prefix:
+            # Strings are right aligned in percent fmt by default, and indicate
+            # left alignment through a negative prefix.
+            #
+            # fstrings left align by default, and separate signs from alignment
+            #
+            # Python even accepts float values here, for both percent fmt
+            # and fstrings
+            #
+            # In order to not have to figure out what sort of number we are
+            # dealing with, just look at the leading - sign (if there is one)
+            # and remove it for the conversion
+            if aggressive and fmt_prefix.startswith("-"):
```

Comment:
```suggestion
            if fmt_prefix.startswith("-"):
```

It seems that we have handled all possible cases within this branch (fmt_spec == "s" and fmt_prefix), so there is no room for error anymore; therefore we don't need aggressive flag?

Is there any known/suspected case where outputs might differ for original and new f-string? to me it sounds exhaustive what you've suggested.

Reply:
The reason I "locked" this behind the aggressive flag is that it changes the current behaviour. Arguably the current behaviour is wrong (a right aligned percentage string would become a left aligned fstring), but still.
If you feel that's not critical I'm fine with dropping the flag requirement.